### PR TITLE
Added KM113 keyboard

### PR DESCRIPTION
**Pull request by yelishang**
## Description

Added KM113 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/jidohun/qmk113/info.json [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": false,
+        "nkro": true,
+        "haptic": false,
+        "rgb_matrix": true,
+        "rgblight": false,
+        "encoder": true
```
Comment (waffle87) [inline keyboards/jidohun/qmk113/info.json:25]:
```suggestion
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgb_matrix": true,
        "encoder": true
```

File: keyboards/jidohun/qmk113/info.json [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        ]
+    },
+    "ws2812": {
+        "pin": "A8"
+    },
+    "rgb_matrix": {
+        "driver": "ws2812",
+        "max_brightness": 108,
+        "val_steps": 27,
+        "speed_steps": 64,
```
Comment (waffle87) [inline keyboards/jidohun/qmk113/info.json:43]: These are quite large jumps, not the most friendly values..

File: keyboards/jidohun/qmk113/rules.mk [hunk 3]
```diff
@@ -0,0 +1,2 @@
+EEPROM_DRIVER = wear_leveling
```
Comment (waffle87) [inline keyboards/jidohun/qmk113/rules.mk:1]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by yelishang** [comment]
Are there any problems that need to be modified or merged

**Comment by yelishang** [comment]
Is it possible to merge.

File: keyboards/jidohun/qmk113/config.h [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+#define SPI_MOSI_PIN B5
+#define SPI_MISO_PIN B4
+#define SPI_MOSI_PAL_MODE 5
+
+/* SPI Flash Config */
+#define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN C12
+#define WEAR_LEVELING_BACKING_SIZE (4 * 1024)
+
+/* RGB Config */
+#define RGB_MATRIX_LED_COUNT 85
```
Comment (waffle87) [inline keyboards/jidohun/qmk113/config.h:18]:
```suggestion
```

**Comment by yelishang** [comment]
Is there anything else that needs to be modified

File: keyboards/jidohun/qmk113/config.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+#define SPI_MOSI_PIN B5
+#define SPI_MISO_PIN B4
+#define SPI_MOSI_PAL_MODE 5
+
+/* SPI Flash Config */
+#define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN C12
+#define WEAR_LEVELING_BACKING_SIZE (4 * 1024)
+
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_KEYPRESSES
```
Comment (drashna) [inline keyboards/jidohun/qmk113/config.h:18]: These are not needed, as they're inferred by the enabled animations. 
```suggestion
```

File: keyboards/jidohun/qmk113/config.h [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+/* SPI Config */
+#define SPI_DRIVER SPIDQ
+#define SPI_SCK_PIN B3
+#define SPI_MOSI_PIN B5
+#define SPI_MISO_PIN B4
+#define SPI_MOSI_PAL_MODE 5
+
+/* SPI Flash Config */
+#define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN C12
+#define WEAR_LEVELING_BACKING_SIZE (4 * 1024)
```
Comment (drashna) [inline keyboards/jidohun/qmk113/config.h:15]:
```suggestion
```

File: keyboards/jidohun/qmk113/info.json [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "encoder": true
+    },
+    "matrix_pins": {
+        "cols": ["C0", "C1", "C2", "C3", "A5", "B1", "B10", "B11", "B12", "B13", "B14", "B15", "C6", "C7", "C8", "C9"],
+        "rows": ["A0", "A1", "A2", "A3", "A4", "C13"]
+    },
+    "encoder": {
```
Comment (drashna) [inline keyboards/jidohun/qmk113/info.json:27]:
```suggestion
    "eeprom": {
        "wear_leveling": {
            "backing_size": 4096,
            "driver": "spi_flash"
        }
    },
    "encoder": {
```

File: keyboards/jidohun/qmk113/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,                  TO(0),    _______,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,                   TO(1),    _______,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                    TO(2),    _______,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             _______, KC_APP, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT, TO(3),    _______
+    )
+};
+
+/* encoder; start */
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (drashna) [inline keyboards/jidohun/qmk113/keymaps/via/keymap.c:47]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/jidohun/qmk113/rules.mk [hunk 9]
```diff
@@ -0,0 +1 @@
+WEAR_LEVELING_DRIVER = spi_flash
```
Comment (drashna) [inline keyboards/jidohun/qmk113/rules.mk:1]:
```suggestion
```

File: keyboards/jidohun/qmk113/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             _______, KC_APP, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT, TO(3),    KC_PGDN
+    ),
+
+    [3] = LAYOUT(
+        KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,                  KC_MUTE,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,                  TO(0),    _______,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,                   TO(1),    _______,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                    TO(2),    _______,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             _______, KC_APP, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT, TO(3),    _______
```
Comment (drashna) [inline keyboards/jidohun/qmk113/keymaps/default/keymap.c:41]: These layers look to be identical to the base layer, unless I'm missing something.   As such, they can/should be removed. 
```suggestion
```

File: keyboards/jidohun/qmk113/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    ),
+
+    [3] = LAYOUT(
+        KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,                  KC_MUTE,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,                  TO(0),    _______,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,                   TO(1),    _______,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,                    TO(2),    _______,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             _______, KC_APP, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT, TO(3),    _______
+    )
```
Comment (drashna) [inline keyboards/jidohun/qmk113/keymaps/via/keymap.c:42]: Same here.
```suggestion
```

File: keyboards/jidohun/qmk113/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+};
+
+/* encoder; start */
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [1] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [2] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [3] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
```
Comment (drashna) [inline keyboards/jidohun/qmk113/keymaps/via/keymap.c:51]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
The board (probably) should be renamed, as this isn't a board released by QMK itself.

https://qmk.fm/powered/

File: keyboards/jidohun/qmk113/keymaps/default/keymap.c (hunk 10 above)
Comment (yelishang) [reply keyboards/jidohun/qmk113/keymaps/default/keymap.c:41]: Layer 2-3 is the difference between the MAC system and the previous two layers, which is that LGUI and LALT are interchangeable and cannot be deleted

File: keyboards/jidohun/qmk113/rules.mk (hunk 9 above)
Comment (yelishang) [reply keyboards/jidohun/qmk113/rules.mk:1]: This is for selecting spi_flash driver is deleted, it does not match the actual situation and cannot be deleted

File: keyboards/jidohun/qmk113/keymaps/via/keymap.c (hunk 11 above)
Comment (yelishang) [reply keyboards/jidohun/qmk113/keymaps/via/keymap.c]: Layer 2-3 is the difference between the MAC system and the previous two layers, which is that LGUI and LALT are interchangeable and cannot be deleted

File: keyboards/jidohun/qmk113/keymaps/via/keymap.c (hunk 12 above)
Comment (yelishang) [reply keyboards/jidohun/qmk113/keymaps/via/keymap.c]: Layer 2-3 is the difference between the MAC system and the previous two layers, which is that LGUI and LALT are interchangeable and cannot be deleted

File: keyboards/jidohun/km113/info.json [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+                { "matrix":[5,2], "x": 2.5, "y": 5, "w": 1.25 },
+                { "matrix":[5,5], "x": 3.75, "y": 5, "w": 6.25 },
+                { "matrix":[5,8], "x": 10, "y": 5 },
+                { "matrix":[5,9], "x": 11, "y": 5 },
+                { "matrix":[5,10], "x": 12, "y": 5 },
+                { "matrix":[5,11], "x": 13.25, "y": 5.25 },
+                { "matrix":[5,12], "x": 14.25, "y": 5.25 },
+                { "matrix":[5,13], "x": 15.25, "y": 5.25 }
+                { "matrix":[5,14], "x": 16.5, "y": 5 }
+                { "matrix":[5,15], "x": 17.5, "y": 5 }
```
Comment (itarze) [inline keyboards/jidohun/km113/info.json:280]:
```suggestion
                { "matrix":[5,13], "x": 15.25, "y": 5.25 },
                { "matrix":[5,14], "x": 16.5, "y": 5 },
                { "matrix":[5,15], "x": 17.5, "y": 5 }
```

**Review by drashna — APPROVED** [review]

**Comment by yelishang** [comment]
May I ask if there are any areas that need to be modified after the merger.

File: keyboards/jidohun/km113/info.json
Comment (fauxpark) [inline keyboards/jidohun/km113/info.json:1]: Please run `qmk format-json -i` on this file.

Comment (yelishang) [reply keyboards/jidohun/km113/info.json:1]: It has been updated with the above command

**Review by fauxpark — APPROVED** [review]

**Comment by yelishang** [comment]
Can we merge now。

File: keyboards/jidohun/km113/info.json [hunk 14]
```diff
@@ -0,0 +1,272 @@
...
+            {"matrix": [3, 0], "x": 0, "y": 38, "flags": 4},
+            {"matrix": [4, 0], "x": 0, "y": 51, "flags": 4},
+            {"matrix": [5, 0], "x": 0, "y": 64, "flags": 4},
+            {"matrix": [5, 1], "x": 15, "y": 64, "flags": 4},
+            {"matrix": [5, 2], "x": 30, "y": 64, "flags": 4},
+            {"matrix": [5, 5], "x": 75, "y": 64, "flags": 4}
+        ],
+        "max_brightness": 108,
+        "speed_steps": 64,
+        "val_steps": 27
```
Comment (zvecr) [inline keyboards/jidohun/km113/info.json:168]: These values are poor choices as default values.

```suggestion
        "max_brightness": 108
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by yelishang** [comment]
Can we merge now。

**Comment by yelishang** [comment]
Do you need any further modifications to merge

**Comment by yelishang** [comment]
Do you need any further modifications to merge？

File: keyboards/jidohun/km113/info.json [hunk 15]
```diff
@@ -0,0 +1,270 @@
...
+            "jellybean_raindrops": true,
+            "multisplash": true,
+            "pixel_flow": true,
+            "pixel_fractal": true,
+            "pixel_rain": true,
+            "rainbow_beacon": true,
+            "rainbow_moving_chevron": true,
+            "rainbow_pinwheels": true,
+            "raindrops": true,
+            "solid_color": true,
```
Comment (fauxpark) [inline keyboards/jidohun/km113/info.json:64]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by yelishang** [comment]
Can we merge now

**Review by zvecr — CHANGES_REQUESTED** [review]
`keyboards/jidohun/km113/rules.mk`  was incorrectly deleted in 5dc636466b21b0321332085a612f5d1f63f1bc50 and the keyboard does not actually compile.

**Comment by yelishang** [comment]
Can merge?

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**